pep-0572.rst

```diff
@@ -144,50 +142,45 @@ There are a few places where assignment expressions are not allowed,
 in order to avoid ambiguities or user confusion:
 
 - Unparenthesized assignment expressions are prohibited at the top
-  level of an expression statement; for example, this is not allowed::
+  level of an expression statement.  Example::
 
     y := f(x)  # INVALID
+    (y := f(x))  # Valid, though not recommended
 
   This rule is included to simplify the choice for the user between an
   assignment statements and an assignment expression -- there is no
   syntactic position where both are valid.
 
 - Unparenthesized assignment expressions are prohibited at the top
-  level in the right hand side of an assignment statement; for
-  example, the following is not allowed::
+  level of the right hand side of an assignment statement.  Example::
 
     y0 = y1 := f(x)  # INVALID
+    y0 = (y1 := f(x))  # Valid, though discouraged
 
   Again, this rule is included to avoid two visually similar ways of
   saying the same thing.
 
 - Unparenthesized assignment expressions are prohibited for the value
-  of a keyword argument in a call; for example, this is disallowed::
+  of a keyword argument in a call.  Example::
 
     foo(x = y := f(x))  # INVALID
+    foo(x=(y := f(x)))  # Valid, though probably confusing
 
   This rule is included to disallow excessively confusing code, and
   because parsing keyword arguments is complex enough already.
 
-- Assignment expressions (even parenthesized or occurring inside other
-  constructs) are prohibited in function default values.  For example,
-  the following examples are all invalid, even though the expressions
-  for the default values are valid in other contexts::
+- Unparenthesized assignment expressions are prohibited at the top
+  level of a function default value.  Example::
 
     def foo(answer = p := 42):  # INVALID
         ...
-
-    def bar(answer = (p := 42)):  # INVALID
-        ...
-
-    def baz(callback = (lambda arg: p := arg)):  # INVALID
+    def foo(answer=(p := 42)):  # Valid, though not great style
```

Comment:
At this point I'll bet an entire dollar we'll see this more than once:

    def foo(arg=(_sentinel = object())):
        if arg is _sentinel:
            # it's certain no argument was passed

And, at first glance, that seems better than "cluttering" the module with `_sentinel = object()` at top level.  But it all falls apart if they include a second function doing the same thing.  In that case it would be better if the bindings of `_sentinel` were all at top level, so the error is more visually obvious.

I wouldn't care, except this is the only use for an assignment expression in this context I've been able to dream up that seems even slightly plausible to me - and, alas, it's a lot more than just slightly plausible.

Not that I want to go back to forbidding it either.  Just setting up an "I told you so!" for later 😉